tests/007-test-setting-extra-opts.sh

```diff
@@ -0,0 +1,68 @@
+source $SRCDIR/libtest.sh
+
+# Test that the user-specified options stored in
+# EXTRA_DOCKER_STORAGE_OPTIONS actually end up in 
+# /etc/sysconfig/docker-storage, appended in the string
+# DOCKER_STORAGE_OPTIONS.
+test_set_extra_docker_opts() {
+  local devs=$TEST_DEVS
+  local test_status
+  local testname=`basename "$0"`
+  local vg_name
+  local vg_name="dss-test-foo"
+  local extra_options="--storage-opt dm.fs=ext4"
+
+  # Error out if any pre-existing volume group vg named dss-test-foo
+  for vg in $(vgs --noheadings -o vg_name); do
+    if [ "$vg" == "$vg_name" ]; then
+      echo "ERROR: $testname: Volume group $vg_name already exists."
+      return 1
+    fi
+  done
+
+  # Create config file with Docker storage options
+  clean_config_files
+  cat << EOF > /etc/sysconfig/docker-storage-setup
+DEVS="$devs"
+VG=$vg_name
+EXTRA_DOCKER_STORAGE_OPTIONS="$extra_options"
+EOF
+  
+  # Run docker-storage-setup
+  $DSSBIN >> $LOGS 2>&1 
+
+  # d-s-s failed
+  if [ $? -ne 0 ]; then 
+    cleanup $vg_name "$devs"
+    return $test_status
+  fi
+  
+  test_status=1
+
+  # Check /etc/sysconfig/docker-storage to see if 
+  # $EXTRA_DOCKER_STORAGE_OPTIONS was appended to
+  # DOCKER_STORAGE_OPTIONS in /etc/sysconfig/docker-storage
+  if [ ! -f /etc/sysconfig/docker-storage ]; then
+    echo "ERROR: $testname: docker-storage config file does not exist"
+    return $test_status
+  fi
+
+  source /etc/sysconfig/docker-storage
+  local options=$DOCKER_STORAGE_OPTIONS
+ 
+  # http://stackoverflow.com/questions/3685970/check-if-an-array-contains-a-value 
+  for opt in $extra_options; do
+    if [[ ! "${options[@]}" =~ "$opt" ]]; then
+      cleanup $vg_name "$devs"
+      return $test_status
+    fi
+  done
+  
```

Comment:
This will test two separate values. dm.fs and ext4. Why not simply use grep  to see if extra_options is part of options or not.